### DISCUSSION TITLE
Embedding files to serverless functions

### DISCUSSION THREAD

**Discussion by aidan-rypens** [vercel/vercel#8754] [Help]
Having some trouble including files (email templates) to serverless functions. Using a Nextjs app + an apollo server that acts as the api.

- Is it possible to embed a file to when the function is spinned up (even with Nextjs)?
- Is it possible to have this file as a private file? So not through the `public` folder?

Some code:

Upon build I can see:
- `.next/server/pages/api.js` is created

Tried the `includeFiles` option in `vercel.json`.

```
{
  "functions": {
    "pages/api/*/**": {
      "includeFiles": "server/emails/templates/*/**"
    }
  }
}
```

Server is a folder in the repo, containing the functions and files needed. What values need to go in the `vercel.json`? Is it where the function is been put (after build, so for example with all the other functions in the same .JS file) or before build, in the seperate .TS file for this function, in this case?

[Documentation](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes#including-additional-files)

Thank you for any insights!

**Comment by aidan-rypens** [comment]
Update: Okay, my bad, didn't place `vercel.json` in the root. Now the CLI is reporting that it is skipped due to it been a NextJS project.

Is there still a way to privately embed files without using?

**Comment by amyegan** [comment]
Hi @AidanRRR! You can follow this guide to import files for use inside a Serverless Function: https://vercel.com/guides/how-can-i-use-files-in-serverless-functions

**Reply by aidan-rypens** [reply]
I've gone through that, but it doesn't seem to 'pack' any of the things that I import in my function. It analyzes this and embeds the files that are required automatically? If so, it doesn't seem to work in my function.

The following code is used in the function:

```
  const emailPath = `${path.join(
    process.cwd(),
    'server/emails/templates',
    'promo.html'
  )}`;

  const html = (await readFile(emailPath, 'utf8')).replace('{{code}}', code);
```

**Reply by amyegan** [reply]
Can you help me understand what about it isn't working for you? If you able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), I would be happy to help with debugging.
